contracts/TreasuryInsurance.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@openzeppelin/contracts/utils/Context.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC721/ERC721.sol';
+import './interfaces/IERC20Allowance.sol';
+import './Treasury.sol';
+
+// Treasury contract with insurance.
+contract TreasuryInsurance is Treasury, ERC721 {
+    mapping(uint256 => uint256) public compLimits;
+    mapping(uint256 => string) public insuranceConditions;
+    mapping(uint256 => uint256) public insuranceCosts;
+    mapping(uint256 => address) public insuranceAssets;
+    mapping(uint256 => uint256) public lastPayTime;
+    mapping(uint256 => uint256) public requestedComps;
+    mapping(uint256 => string) public requestedCases;
+
+    uint256 public tickets;
```

Comment:
question: @shahankhatch what nomenclature do we want for insurance?

My experience as a consumer of insurance products I've encountered `policy`, `policy_number` or `policy_id`, where as the term `tickets` is prevalent in gaming systems (e.g. lotteries and some consensus mechanisms it turns out 😁).

_This question is probably something to think about, maybe gather consensus with the committee (I think it'll make sense to align terminology used in our public repos with the corresponding marketing / biz terms)_